protocol_details/rpmsg_comms.rst

```diff
@@ -2,28 +2,95 @@
+  future transmissions from the main controller.
+* Optionally triggera an interrupt to inform main controller of freed buffer.
+
+  * if F_NO_NOTIFY flag is not set in used ring buffer
+
 
 .. image:: ../images/rpmsg_flow.jpg
 
-The remote core gets the received RPMsg buffer from the “avail” ring buffer, processes it and then
-returns it back to the “used” ring buffer. When the remote core is sending a message to the main
-controller, “avail” and “used” ring buffers role are swapped.
-
-The reason for swapping the roles of the ring buffers comes from the fact, that the main core
-works as a Buffer Provider. The Buffer Provider has complete control of memory management and
-shared memory allocation. Obviously, when the main controller, or Buffer Provider, does not fill
-the “avail” ring buffer of VRING0 (Orange), the Remote core is unable to send a message to the
-main controller.
-This can be used to throttle the communication generated by the Remote core. It is to be noticed,
-that the main controller always dequeues from the “used” ring buffer and enqueues to the “avail”
-ring buffer.
-For the remote, the situation is inverse.
-
-The triggering of interrupts is optional. It is governed
-by the flags in “used” and “avail” ring buffers.
+
+Receive RPMsg flow (remote to main)
+-----------------------------------
+
+For the transmission of RPMsgs from the remote to the main controller (receive flow) the
+following actions are performed on the receive VRING (VRING0), as detailed in the lower part
+of the flow chart above.
+
+The remote
+
+* Gets a transmission buffer from the available queue of VRING0.
+  * The main controller can throttle remote messages by withholding the available buffer.
+* Writes the RPMsg header and payload of this buffer.
+* Enqueues this buffer containing the RPMsg to the used queue of VRING0 by setting the head
+  index.
+* Optionally triggers an interrupt to inform the main controller of message availability.
+
+  * if F_NO_NOTIFY flag is not set in used ring buffer.
+
+The main controller then
+
+* Gets the received buffer from the used queue of VRING0.
+* Passes the buffer to the endpoint callback, which would handle the message.
+* Enqueues the freed buffer to the available queue of VRING0, to make it available again for
+  future transmissions from the remote.
+* Optionally triggers an interrupt to inform the remote of freed buffer.
+
+  * if F_NO_NOTIFY flag is not set in used ring buffer.
+
+Summary
+-------
+
+The remote gets the received RPMsg buffer from the available ring buffer, processes it and
+then returns it back to the used ring buffer. When the remote is sending a message to the
+main controller, available and used ring buffer roles are swapped.
+
+The reason for swapping the roles of the ring buffers stems from the main controller being the
+buffer provider. The buffer provider has complete control of memory management and shared memory
```

Comment:
"The buffer provider has complete control of memory management" is not exact. vs hypervisor and co.
Here you could just say `The main  controller is in charge of  the shared  memory allocation.`